qiskit/algorithms/eigensolvers/eigensolver.py

```diff
@@ -0,0 +1,105 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Eigensolver interface"""
+
+from abc import ABC, abstractmethod
+from typing import Optional, List, Tuple
+
+import numpy as np
+
+from qiskit.opflow import OperatorBase
+from ..algorithm_result import AlgorithmResult
+from ..list_or_dict import ListOrDict
+
+
+class Eigensolver(ABC):
+    """The Eigensolver Interface.
+
+    Algorithms that can compute eigenvalues for an operator
+    may implement this interface to allow different algorithms to be
+    used interchangeably.
+    """
+
+    @abstractmethod
+    def compute_eigenvalues(
+        self, operator: OperatorBase, aux_operators: Optional[ListOrDict[OperatorBase]] = None
+    ) -> "EigensolverResult":
+        """
+        Computes eigenvalues. Operator and aux_operators can be supplied here and
+        if not None will override any already set into algorithm so it can be reused with
+        different operators. While an operator is required by algorithms, aux_operators
+        are optional. To 'remove' a previous aux_operators array use an empty list here.
+
+        Args:
+            operator: Qubit operator of the Observable
+            aux_operators: Optional list of auxiliary operators to be evaluated with the
+                eigenstate of the minimum eigenvalue main result and their expectation values
+                returned. For instance in chemistry these can be dipole operators, total particle
+                count operators so we can get values for these at the ground state.
+
+        Returns:
+            EigensolverResult
+        """
+        return EigensolverResult()
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        """Whether computing the expectation value of auxiliary operators is supported.
+
+        Returns:
+            True if aux_operator expectations can be evaluated, False otherwise
+        """
+        return False
+
+
+class EigensolverResult(AlgorithmResult):
+    """Eigensolver Result."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._eigenvalues = None
+        self._eigenstates = None
+        self._aux_operator_eigenvalues = None
+
+    @property
+    def eigenvalues(self) -> Optional[np.ndarray]:
+        """returns eigen values"""
+        return self._eigenvalues
+
+    @eigenvalues.setter
+    def eigenvalues(self, value: np.ndarray) -> None:
+        """set eigen values"""
+        self._eigenvalues = value
+
+    @property
+    def eigenstates(self) -> Optional[np.ndarray]:
+        """return eigen states"""
+        return self._eigenstates
+
+    @eigenstates.setter
+    def eigenstates(self, value: np.ndarray) -> None:
+        """set eigen states"""
+        self._eigenstates = value
+
```

Comment:
I do not believe we are going to be able to generate states back in general. We can from the NumPy one, but will no longer be able to do so from VQD. We used to infer we would have states if the aux_operators were supported since we needed to be able to compute/get access to state for these. Hence we could pass it back. But this is no longer the case with Estimator where aux_ops are supported but we have no access to state. Maybe users will just have to check whether this passes back None or not. Maybe for this, VQE etc perhaps we should no longer have state in base class.

Reply:
Precisely, retrieving the eigenstate would require passing on an additional sampler and executing a sampler job, I would agree with removing the eigenstates from the VQE base class. At least in VQD, I am returning an empty list.

Reply:
@woodsp-ibm I have removed `eigenstates` from `EigensolverResult`, and added a quick fix to still be able to evaluate the auxiliary operators for the numpy eigensolver (it used `result.eigenstates`). As far as I have seen, this is consistent with the changes in `minimum_eigensolvers`.

Reply:
I think this is consistent with what is being done for the MinimumEigensolvers and the numpy one there in #8702 Yes we would want to add this new result class to the docs.